x-pack/elastic-agent/pkg/remote/client.go

```diff
@@ -272,17 +277,18 @@ func prefixRequestFactory(URL string) requestFunc {
 }
 
 // makeTransport create a transport object based on the TLS configuration.
-func makeTransport(timeout time.Duration, tls *tlscommon.Config) (*http.Transport, error) {
+func makeTransport(scheme string, timeout time.Duration, tls *tlscommon.Config) (http.RoundTripper, error) {
+	dialer := transport.NetDialer(timeout)
+	if scheme == "http" {
+		return &http.Transport{Dial: dialer.Dial}, nil
+	}
 	tlsConfig, err := tlscommon.LoadTLSConfig(tls)
```

Comment:
this was first just so you dont end up instantiating dialer in case tls config is incorrect.